pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -159,18 +149,26 @@ func getMonitoringConsoleDeployment(cr splcommon.MetaObject, spec *enterprisev1.
 		},
 	}
 
-	env := []corev1.EnvVar{
-		{Name: "SPLUNK_CONFIGMAP_HASH",
-			Value: configMapHash,
-		},
-	}
+	env := []corev1.EnvVar{}
+
 	// append labels and annotations from parent
-	splcommon.AppendParentMeta(deploymentMC.Spec.Template.GetObjectMeta(), cr.GetObjectMeta())
+	splcommon.AppendParentMeta(statefulSet.Spec.Template.GetObjectMeta(), cr.GetObjectMeta())
 
 	// update statefulset's pod template with common splunk pod config
-	updateSplunkPodTemplateWithConfig(&deploymentMC.Spec.Template, cr, spec, instanceType, env, secretName)
+	updateSplunkPodTemplateWithConfig(client, &statefulSet.Spec.Template, cr, spec, instanceType, env, secretName)
 
-	return deploymentMC, nil
+	//update podTemplate annotation with configMap resource version
+	var monitoringConsoleConfigMap *corev1.ConfigMap
+	if cr.GetObjectMeta().GetDeletionTimestamp() != nil {
+		monitoringConsoleConfigMap, _ = ApplyMonitoringConsoleEnvConfigMap(client, cr.GetNamespace(), cr.GetName(), env, false)
```

Comment:
Use GetConfigMapByName API after PR https://github.com/splunk/splunk-operator/pull/134 is merged